components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx

```diff
@@ -0,0 +1,690 @@
+  isSwapTransaction,
+  getETHSwapTransactionBuyAndSellTokens,
+  getTransactionTransferredValue,
+  getIsTxApprovalUnlimited
+} from '../../../utils/tx-utils'
+import {
+  accountInfoEntityAdaptorInitialState
+} from '../../../common/slices/entities/account-info.entity'
+import {
+  makeNetworkAsset
+} from '../../../options/asset-options'
+import {
+  getCoinFromTxDataUnion
+} from '../../../utils/network-utils'
+import {
+  getAddressLabel,
+  getAccountLabel
+} from '../../../utils/account-utils'
+import {
+  computeFiatAmount
+} from '../../../utils/pricing-utils'
+import {
+  isNativeAsset
+} from '../../../utils/asset-utils'
+import Amount from '../../../utils/amount'
+
+// Hooks
+import {
+  useGetAccountInfosRegistryQuery,
+  useGetDefaultFiatCurrencyQuery,
+  useGetNetworkQuery,
+  useGetTokenSpotPricesQuery,
+} from '../../../common/slices/api.slice'
+import {
+  useAccountQuery,
+  useGetCombinedTokensListQuery
+} from '../../../common/slices/api.slice.extra'
+
+// Components
+import {
+  NftIcon
+} from '../../shared/nft-icon/nft-icon'
+import {
+  Skeleton
+} from '../../shared/loading-skeleton/styles'
+import {
+  withPlaceholderIcon
+} from '../../shared/create-placeholder-icon'
+import {
+  TransactionDetailsModal
+} from '../popup-modals/transaction_details_modal/transaction_details_modal'
+import {
+  CreateNetworkIcon
+} from '../../shared/create-network-icon'
+
+// Styled Components
+import {
+  PortfolioTransactionItemWrapper,
+  TransactionTypeIcon,
+  DateText,
+  TransactionTypeText,
+  IntentAddressText,
+  AssetIcon,
+  SwapIcon,
+  SwapSellIcon,
+  SwapBuyIcon,
+  TokenNameText,
+  TokenSymbolText,
+  AssetValueText,
+  FiatValueText,
+  ArrowIcon,
+  ArrowIconWrapper,
+  StatusBubble,
+  StatusIcon,
+  LoadingIcon,
+  BalancesColumn,
+  NetworkIconWrapper,
+  IconWrapper,
+  SwapIconsWrapper,
+  SellIconPlaceholder,
+  BuyIconPlaceholder,
+  SwapPlaceholderIcon
+} from './portfolio_tranasction_item.style'
+import {
+  Column,
+  Row,
+  VerticalSpace
+} from '../../shared/style'
+
+export interface Props {
+  transaction: BraveWallet.TransactionInfo | SerializableTransactionInfo
+  isFocused?: boolean
+}
+
+const ICON_ASSET_CONFIG =
+  { size: 'medium', marginLeft: 0, marginRight: 0 } as const
+const ICON_SWAP_CONFIG =
+  { size: 'small', marginLeft: 0, marginRight: 0 } as const
+const ICON_NFT_CONFIG =
+  { size: 'medium', marginLeft: 0, marginRight: 0 } as const
+const AssetIconWithPlaceholder =
+  withPlaceholderIcon(AssetIcon, ICON_ASSET_CONFIG)
+const SwapIconWithPlaceholder =
+  withPlaceholderIcon(SwapIcon, ICON_SWAP_CONFIG)
+const NftIconWithPlaceholder =
+  withPlaceholderIcon(NftIcon, ICON_NFT_CONFIG)
+
+export const PortfolioTransactionItem =
+  React.forwardRef<HTMLDivElement, Props>(({
+    transaction,
+    isFocused
+  }: Props, forwardedRef) => {
+
+    // State
+    const [showTransactionDetails, setShowTransactionDetails]
+      = React.useState<Boolean>(false)
```

Comment:
unneeded and incorrect type annotation. (Should be `boolean`, not `Boolean`)

Reply:
Updated